### PR TITLE
GeocentricConverter equality check after WGS param overrides

### LINKED ISSUE #32: Reprojecting British National Grid (27700) coords to Mercator (3857) using Proj4j
On epsg.io https://epsg.io/transform#s_srs=27700&t_srs=3857&x=327420.9886680&y=690284.5471100 the BNG easting/northing coodinates 327420.988668 690284.547110 reproject correctly to -352695.04 7578309.23. If I attempt to reproject the same coordinates using proj4j (0.1.0) with the following code:

```
Double easting = 327420.988668;
Double northing = 690284.54711;
CoordinateReferenceSystem BNG = crsFactory.createFromParameters("EPSG:27700", "+proj=tmerc +lat_0=49 +lon_0=-2 +k=0.9996012717 +x_0=400000 +y_0=-100000 +ellps=airy +towgs84=446.448,-125.157,542.06,0.15,0.247,0.842,-20.489 +units=m +no_defs");
CoordinateReferenceSystem MERCATOR = crsFactory.createFromName("EPSG:3857");
CoordinateTransform trans = ctFactory.createTransform(BNG, MERCATOR);
ProjCoordinate in = new ProjCoordinate(easting, northing);
ProjCoordinate out = new ProjCoordinate();
trans.transform(in, out);
log.info(String.format("%s : %s", out.x, out.y));
```
I get -352695.04030562507 : 7542788.088927755. The Y coord is around 30 miles south of the correct position. If I use version 1.0.0 of proj4j, I get -352535.43434034986 : 7578324.217405743, which is much closer but still around 200 meters out.

Incidently if I reproject first to EPSG:4326 then to EPSG:3857, I get the correct result.

### PR DESCRIPTION
**Pull request by bosborn**
* GeocentricConverter equality check after WGS param overrides
* Previously failing tests
  * EPSG:27700
  * EPSG:31469
  * EPSG:2056
  * EPSG:4055

FIXES #32

### CODE REVIEW

**Comment by bosborn** [comment]
Any thoughts on the updates to this PR and/or the current state?

**Comment by pomadchin** [comment]
@bosborn oops just felt off my sight; will review it shortly 👍

File: src/main/java/org/locationtech/proj4j/datum/GeocentricConverter.java [hunk 1]
```diff
@@ -60,15 +60,16 @@
...
     }
 
-    public GeocentricConverter(double a, double b) {
+    public GeocentricConverter(double a, double b, double e2) {
         this.a = a;
         this.b = b;
         a2 = a * a;
         b2 = b * b;
-        e2 = (a2 - b2) / a2;
+        this.e2 = e2;
```
Comment (pomadchin) [inline src/main/java/org/locationtech/proj4j/datum/GeocentricConverter.java:72]: `e2 = ellipsoid.getEccentricitySquared()`

**Review by pomadchin — APPROVED** [review]
🚀 thanks for leaving comments all over the places, totally makes a lot of sense. Thanks for a solid contribution!

**Comment by pomadchin** [comment]
Let me know when you need a release with this fix

**Comment by bosborn** [comment]
Excellent, thanks.  Whenever you get the chance I could use a new release.

**Pull request merged**